### ISSUE TITLE
Panic in txpool with 2 sentries

### ISSUE THREAD

**Issue by yperbasis** [ledgerwatch/erigon#4679]
When I set up 2 sentries (`backend.sentryServers` in eth/backend.go), after some time Erigon crashes with
```
[INFO] [07-01|11:51:03.646] [1/16 Headers] Inserting headers         progress=14739189 queue=1
panic: interface conversion: interface {} is nil, not *simplelru.entry

goroutine 3496 [running]:
github.com/hashicorp/golang-lru/simplelru.(*LRU).removeElement(0x17b61e0?, 0xc0612b5660?)
        github.com/hashicorp/golang-lru@v0.5.5-0.20210104140557-80c98217689d/simplelru/lru.go:172 +0x107
github.com/hashicorp/golang-lru/simplelru.(*LRU).removeOldest(0x1693540?)
        
github.com/hashicorp/golang-lru@v0.5.5-0.20210104140557-80c98217689d/simplelru/lru.go:165 +0x34
github.com/hashicorp/golang-lru/simplelru.(*LRU).Add(0xc073811d80, {0x161a440?, 0xc0c67ab5c0}, {0x1664a60?, 0x2a7af50})
        github.com/hashicorp/golang-lru@v0.5.5-0.20210104140557-80c98217689d/simplelru/lru.go:67 +0x394
github.com/ledgerwatch/erigon-lib/txpool.(*TxPool).discardLocked(0xc08c3baf00, 0xc0c6b4dcc0, 0xe)
        github.com/ledgerwatch/erigon-lib@v0.0.0-20220629154434-59f7b5b57b68/txpool/pool.go:1077 +0x15a
github.com/ledgerwatch/erigon-lib/txpool.promote(0xc06aab7bc0, 0xc06aab7c00, 0xc06aab7c40, 0x0, 0xc0612b5be8)
        github.com/ledgerwatch/erigon-lib@v0.0.0-20220629154434-59f7b5b57b68/txpool/pool.go:1310 +0x394
github.com/ledgerwatch/erigon-lib/txpool.addTxs(0x0, {0x1f27260, 0xc0c6b18c78}, 0x1152?, {{0xc0b802c600, 0x2c, 0x40}, {0xc0c646b200, 0x370, 0x480}, ...}, ...)
        github.com/ledgerwatch/erigon-lib@v0.0.0-20220629154434-59f7b5b57b68/txpool/pool.go:937 +0x787
github.com/ledgerwatch/erigon-lib/txpool.(*TxPool).processRemoteTxs(0xc08c3baf00, {0x1f30168, 0xc000718300?})
        github.com/ledgerwatch/erigon-lib@v0.0.0-20220629154434-59f7b5b57b68/txpool/pool.go:510 +0x48d
github.com/ledgerwatch/erigon-lib/txpool.MainLoop({0x1f30168?, 0xc000718300}, {0x1f36ef0, 0xc05750a460}, {0x20?, 0x3fb2b6d0c?}, 0xc08c3baf00, 0xc0738627e0, 0xc06aab7c80, 0xc073801ec0, ...)
        github.com/ledgerwatch/erigon-lib@v0.0.0-20220629154434-59f7b5b57b68/txpool/pool.go:1343 +0x54e
created by github.com/ledgerwatch/erigon/eth.New
        github.com/ledgerwatch/erigon/eth/backend.go:477 +0x3031
```

**Comment by AskAlexSharov** [comment]
Great. Now we know that need 2 sentries to reproduce it. 
Need run txpool with -race to find reason

**Comment by holiman** [comment]
https://github.com/ledgerwatch/erigon-lib/blob/f0140de612446011db101d28a217b347827f2dc4/txpool/pool.go#L579 

`RLock` around `Get` is insufficient, because `Get` updates the recentness. 

Same as 

https://github.com/ledgerwatch/erigon/issues/3799
https://github.com/ledgerwatch/erigon/issues/4396

I made the same mistake in go-ethereum recently. Fix: https://github.com/ethereum/go-ethereum/pull/26164

**Comment by yperbasis** [comment]
> https://github.com/ledgerwatch/erigon-lib/blob/f0140de612446011db101d28a217b347827f2dc4/txpool/pool.go#L579
> 
> `RLock` around `Get` is insufficient, because `Get` updates the recentness.
> 
> Same as
> 
> #3799 #4396
> 
> I made the same mistake in go-ethereum recently. Fix: [ethereum/go-ethereum#26164](https://github.com/ethereum/go-ethereum/pull/26164)

Oh, I see. We'll fix it. @holiman thank you for pointing this out!
